applications: nrf5340_audio: Add DFU support. (#7853)
nrfconnect/sdk-nrf: applications/nrf5340_audio/README.rst

Since the application architecture is uniform and the firmware code is shared, the set of audio modules in use depends on the chosen stream mode (BIS or CIS), the chosen audio inputs and outputs (USB or analog jack), and if the gateway or the headset configuration is selected.

.. note::
   In the current version of the application, bootloader is disabled by default.

>    Device Firmware Update (DFU) can only be enabled when :ref:`nrf53_audio_app_building_script`.

Reviewer: ```suggestion
   Device Firmware Update (DFU) can only be enabled when :ref:`nrf53_audio_app_building_script`.
   See :ref:`nrf53_audio_app_configuration_configure_fota` for details.
```

Since the application architecture is uniform and the firmware code is shared, the set of audio modules in use depends on the chosen stream mode (BIS or CIS), the chosen audio inputs and outputs (USB or analog jack), and if the gateway or the headset configuration is selected.

.. note::

>    In the current version of the application, bootloader is disabled by default.

Reviewer 2: ```suggestion
   In the current version of the application, the bootloader is disabled by default.
```

When testing the application, an additional audio jack cable is required to use I2S.
Use this cable to connect the audio source (PC) to the analog **LINE IN** on the development kit.

.. _nrf53_audio_app_configuration_configure_fota:

Configuring FOTA upgrades
=========================

You can configure FOTA upgrades to replace the applications on both the application core and the network core.

> The nRF5340 Audio application supports the following types of the DFU flash layouts:

Reviewer 2: ```suggestion
The nRF5340 Audio application supports the following types of DFU flash memory layouts:
```

> * Internal flash layout - which supports only single-image DFU.
> * External flash layout - which supports :ref:`multi-image DFU <ug_nrf5340_multi_image_dfu>`.

Reviewer 2: ```suggestion
* Internal flash memory layout - which supports only single-image DFU.
* External flash memory layout - which supports :ref:`multi-image DFU <ug_nrf5340_multi_image_dfu>`.
```

> LE Audio Controller Subsystem for nRF53 supports both the normal size and the minimal size of bootloader.

Reviewer 2: ```suggestion
The LE Audio Controller Subsystem for nRF53 supports both the normal and minimal sizes of the bootloader.
```

The minimal size is specified using the :kconfig:option:`CONFIG_NETBOOT_MIN_PARTITION_SIZE`.

> Hardware requirements for external flash DFU

Reviewer 2: ```suggestion
Hardware requirements for external flash memory DFU
```

--------------------------------------------

> To enable the external flash DFU, you need an additional flash shield.
> The nRF5340 Audio application uses MX25R6435F as the SPI NOR Flash.

Reviewer 2: ```suggestion
To enable the external flash DFU, you need an additional flash memory shield.
The nRF5340 Audio application uses the MX25R6435F as the SPI NOR Flash.
```

* Core type (``-c`` parameter): ``app``, ``net``, or ``both``
* Application version (``-b`` parameter): either ``release`` or ``debug``
* Device type (``-d`` parameter): ``headset``, ``gateway``, or ``both``
* DFU type (``-m`` parameter): ``internal``, ``external``
* Network core bootloader minimal size (``-M``)

See the following examples of the parameter usage with the command run from the :file:`buildprog` directory:

* Example 1: The following command builds the application using the script for the application core with the ``debug`` application version for both the headset and the gateway:

  .. code-block:: console

     python buildprog.py -c app -b debug -d both

* Example 2: The following command builds the application as in the example 1, but with the DFU internal flash layout enabled and the minimal size of the network core bootloader:

  .. code-block:: console

     python buildprog.py -c app -b debug -d both -m internal -M

  If you run this command with the ``external`` DFU type parameter instead, the external flash layout will be enabled with the minimal size of the network core bootloader.

> The command can be ran from any location, as long as the correct path to :file:`buildprog.py` is given.

Reviewer: ```suggestion
The command can be run from any location, as long as the correct path to :file:`buildprog.py` is given.
```

* Core type (``-c`` parameter): ``app``, ``net``, or ``both``
* Application version (``-b`` parameter): either ``release`` or ``debug``
* Device type (``-d`` parameter): ``headset``, ``gateway``, or ``both``
* DFU type (``-m`` parameter): ``internal``, ``external``
* Network core bootloader minimal size (``-M``)

See the following examples of the parameter usage with the command run from the :file:`buildprog` directory:

* Example 1: The following command builds the application using the script for the application core with the ``debug`` application version for both the headset and the gateway:

  .. code-block:: console

     python buildprog.py -c app -b debug -d both

> * Example 2: The following command builds the application as in the example 1, but with the DFU internal flash layout enabled and the minimal size of the network core bootloader:

Reviewer 2: ```suggestion
* Example 2: The following command builds the application as in *example 1*, but with the DFU internal flash memory layout enabled and using the minimal size of the network core bootloader:
```

  .. code-block:: console

     python buildprog.py -c app -b debug -d both -m internal -M

>   If you run this command with the ``external`` DFU type parameter instead, the external flash layout will be enabled with the minimal size of the network core bootloader.

Reviewer 2: ```suggestion
  If you run this command with the ``external`` DFU type parameter instead, the external flash memory layout will be enabled using the minimal size of the network core bootloader.
```

   To obtain access to the repository, `contact the sales department <Contact Us_>`_.

.. |net_core_hex_note| replace:: The network core for both gateway and headsets is programmed with the precompiled Bluetooth Low Energy Controller binary file :file:`ble5-ctr-rpmsg_<XYZ>.hex`, where ``<XYZ>`` corresponds to the controller version, for example :file:`ble5-ctr-rpmsg_3216.hex`.

>    This file includes the LE Audio Controller Subsystem for nRF53 and is provided in the :file:`applications/nrf5340_audio/bin` directory. If DFU is enabled, Bluetooth Low Energy Controller binary file will be :file:`pcft_CPUNET.hex` .

Reviewer: ```suggestion
   This file includes the LE Audio Controller Subsystem for nRF53 and is provided in the :file:`applications/nrf5340_audio/bin` directory.
   If :ref:`DFU <nrf53_audio_app_configuration_configure_fota>` is enabled, the subsystem's binary file will be :file:`pcft_CPUNET.hex`.
```
